### PR TITLE
fix: rename flux-lsp to flux_lsp

### PR DESCRIPTION
**Pull request by nathanielc**
Using `flux-lsp` as the name of the config made it more difficult than
needed to install the config as you needed to escape the dash. All other
configs are named with underscores so this brings the config inline.

Additionally provided a better root_dir function for cases where a git
repository may not exist.

### CODE REVIEW

File: lua/lspconfig/server_configurations/flux_lsp.lua [hunk 1]
```diff
@@ -4,7 +4,9 @@ return {
   default_config = {
     cmd = { 'flux-lsp' },
     filetypes = { 'flux' },
-    root_dir = util.find_git_ancestor,
+    root_dir = function(fname)
+        return util.find_git_ancestor(fname) or vim.fn.getcwd()
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply lua/lspconfig/server_configurations/flux_lsp.lua]: Thanks for the prompt feedback. Changed to use `single_file_support`, much simpler now.

**Comment by mjlbach** [comment]
These should be two separate commits

**Comment by nathanielc** [comment]
@mjlbach Two separate commits? Or two separate PRs happy to do either.

**Comment by mjlbach** [comment]
Two separate commits is fine. Also this is a breaking change so it will need to be announced (note to myself). It is inline with the naming guidelines I added to contributing. I thought about adding a "shim" layer which remaps deprecated names, but it's a 1-liner for most people's configurations so not sure it is worth it.

**Comment by mjlbach** [comment]
I also just checked and this is the only language server with a dash.

**Comment by nathanielc** [comment]
@mjlbach Thanks, two commits now. Yeah I agree this is a small enough breaking change on a small enough language that an announcement is enough.

**Pull request merged**